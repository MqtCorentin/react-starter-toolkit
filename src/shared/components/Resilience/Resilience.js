import React from 'react';
import ResilienceSubstitut from './ResilienceSubstitut';

const Resilience = ({ anomaly, resilienceMode, resilienceModifier, FallbackComponent, children }) =>
  anomaly ? (
    <ResilienceSubstitut
      anomaly={anomaly}
      resilienceMode={resilienceMode}
      resilienceModifier={resilienceModifier}
      FallbackComponent={FallbackComponent}
    />
  ) : (
    children
  );

export default Resilience;
