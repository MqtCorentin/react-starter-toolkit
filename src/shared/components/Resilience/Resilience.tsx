import { ComponentProps, ReactNode } from 'react';
import ResilienceSubstitut from './ResilienceSubstitut';

type TResilience = Omit<ComponentProps<typeof ResilienceSubstitut>, 'anomaly'> & {
  anomaly?: ComponentProps<typeof ResilienceSubstitut>['anomaly'] | null;
  children?: ReactNode;
};

const Resilience = ({ resilienceMode, resilienceModifier, FallbackComponent, anomaly = null, children = null }: TResilience) =>
  anomaly ? (
    <ResilienceSubstitut
      anomaly={anomaly}
      resilienceMode={resilienceMode}
      resilienceModifier={resilienceModifier}
      FallbackComponent={FallbackComponent}
    />
  ) : (
    <>{children}</>
  );

export default Resilience;
