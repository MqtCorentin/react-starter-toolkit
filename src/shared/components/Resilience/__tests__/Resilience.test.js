import { render } from '@testing-library/react';
import Resilience from '../Resilience';

describe('Resilience', () => {
  it('Should return Wrapped Component When use component Resilience and not exist anomaly', () => {
    const CustomComponent = () => <div>No anomaly</div>;
    const baseProps = { loaderMode: 'none' };
    const EnhancedComponent = () => (
      <Resilience>
        <CustomComponent {...baseProps} />
      </Resilience>
    );
    const { getByText } = render(<EnhancedComponent />);
    expect(getByText('No anomaly')).toBeDefined();
  });
  it('Should return Resilience with props When use HOC withResilience and anomaly', () => {
    const CustomComponent = () => <div>No Anomaly</div>;
    const ErrorComponent = () => <div>There is an error</div>;
    CustomComponent.displayName = 'CustomName';
    const baseProps = { loaderMode: 'none' };
    const EnhancedComponent = () => (
      <Resilience anomaly={{ code: 500 }} resilienceMode="fallback" FallbackComponent={ErrorComponent}>
        <CustomComponent {...baseProps} />
      </Resilience>
    );
    const { getByText } = render(<EnhancedComponent />);
    expect(getByText('There is an error')).toBeDefined();
  });
});
