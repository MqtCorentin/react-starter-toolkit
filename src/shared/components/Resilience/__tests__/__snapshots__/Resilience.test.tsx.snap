// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resilience Should render null when Resilience have been called with no anomaly and no child 1`] = `<DocumentFragment />`;
